mantidproject/mantid
# Exclude masked data in 1D plots of MDWorkspaces

Masked data should be omitted from 1D plots.

**To Test:**

Create and plot an MDEW and MDHW with the script below. The MDEW is masked between -1 and 1 and therefore the line should join non-masked data points on either side of this region. The MDHW is masked between -2 and -1 and therefore the x axis should start at -1.

You may notice that between 1 and 2, where data is unmasked for both workspaces, the plots look dissimilar in the shape of the peaks. This will be addressed in a separate issue #15082.

```
ws_mdhw = CreateMDWorkspace(Dimensions='1', Extents='-2,2', Names='h',
                       Units='rlu', SplitInto='4')
FakeMDEventData(ws_mdhw, UniformParams='1000', RandomizeSignal='1')
ws_mdhw = BinMD(ws_mdhw,AlignedDim0='h,-2,2,64')
MaskMD(Workspace=ws_mdhw,Dimensions="h",Extents="-2,-1")

ws_mdew = CreateMDWorkspace(Dimensions='1', Extents='-2,2', Names='h',
                       Units='rlu', SplitInto='64')
FakeMDEventData(ws_mdew, UniformParams='1000', RandomizeSignal='1')
ws_mdew = SliceMD(ws_mdew,AlignedDim0='h,-2,2,64')
MaskMD(Workspace=ws_mdew,Dimensions="h",Extents="-1,1")

plotMD(ws_mdhw,normalization=2, error_bars=True)
plotMD(ws_mdew,normalization=2, error_bars=True)
```
